experimental/Schemes/src/FunctionFields.jl

```diff
@@ -78,7 +78,7 @@ represented by
 function function_field(X::AbsCoveredScheme; check::Bool=true)
   return get_attribute!(X, :function_field) do
     VarietyFunctionField(X, check=check)
-  end
+  end::VarietyFunctionField  # TODO: make the type fully concrete
```

Comment:
Thanks for the attempt! 

However, I'm afraid that making the type fully concrete is a bit too much to ask here. We do not have the getters to deduce all the type parameters from the type of the scheme. For instance, there is one chosen chart `U` of `X` which is used to represent elements of this field. Even the type of the `ambient_polynomial_ring` of `U` can not easily be deduced from `X`. But it is used in the type parameters for the function field.

Reply:
```suggestion
  end::VarietyFunctionField  # can't make this more concrete just using the type of X
```